packages/react-admin-core/src/table/useTableQueryFilter.tsx

```diff
@@ -37,7 +40,7 @@ export function useTableQueryFilter<FilterValues extends AnyObject>(
                 if (!isEqual(filters, newValues)) {
                     setFilters(newValues);
                     if (options.pagingApi) {
```

Comment:
Das bei einem 'page-change' raufgescrollt wird sollte glaub ich immer so sein. Derzeit wird aber bei jedem 'filter-change' auch ein 'page-change' auf page 1 getriggert, quasi ein pager-reset. Das ist aber nicht notwendig bei einem 'filter-change' wenn sich der user ohnehin auf page 1 befindet.

Wenn man auf page 1 ist sollte changePage zu page 1 nicht aufgerufen werden, @nsams ? Das ist das eigentliche Problem hier denke ich. 

```suggestion
                    if (options.pagingApi && options.pagingApi.current !== 1) {
```

Reply:
> Wenn man auf page 1 ist sollte changePage zu page 1 nicht aufgerufen werden, @nsams ? Das ist das eigentliche Problem hier denke ich.

Wenn sich der User aber z.B. auf Seite 3 befindet und einen Filter ändert, dann muss man trotzdem auf Seite 1 zurückspringen und hat dasselbe Problem.

Reply:
> Das bei einem 'page-change' raufgescrollt wird sollte glaub ich immer so sein. Derzeit wird aber bei jedem 'filter-change' auch ein 'page-change' auf page 1 getriggert, quasi ein pager-reset. Das ist aber nicht notwendig bei einem 'filter-change' wenn sich der user ohnehin auf page 1 befindet.
> 
> Wenn man auf page 1 ist sollte changePage zu page 1 nicht aufgerufen werden, @nsams ? Das ist das eigentliche Problem hier denke ich.

Stimmt, das könnten wir uns sparen. Allerdings besteht das problem immer noch beim ersten mal filtern wenn man nicht auf page 1 ist. (Was aber sehr selten passieren wird)


Zu deinem change: current ist ein generic (https://github.com/vivid-planet/react-admin/blob/master/packages/react-admin-core/src/table/useTableQueryPaging.tsx#L5) das ist nicht zwingend eine number. Es müsste mit init verglichen werden

Reply:
Ok, dann muss man aber eigentlich bei einem 'filter-change' nie page-scrollen, egal ob man auf seite 1 oder 3 ist, weil  der 'filter-change' ja über das UI getriggert wird, dh der User ja die Filter eh sieht, Könnt ich mir dass so vorstellen, ohne options:

```ts
    options.pagingApi.changePage(options.pagingApi.init, 1, { noScrollToTop: true });
```

passt aber auch mit options, glaub aber das man das nicht braucht.



> > Die Filter werden wohl nie unter der Tabelle sein, raufscrollen wird also nicht notwendig sein. Das ist nur beim paging notwendig so wie ich das sehe.
> > Bitte änder das fix, ohne einstellmöglichkeit.
> 
> @nsams Meinst du so? [5e851b0](https://github.com/vivid-planet/react-admin/commit/5e851b0bbacaf5b59c8f85c1d0eb06ff70390784)



> > Die Filter werden wohl nie unter der Tabelle sein, raufscrollen wird also nicht notwendig sein. Das ist nur beim paging notwendig so wie ich das sehe.
> > Bitte änder das fix, ohne einstellmöglichkeit.
> 
> @nsams Meinst du so? [5e851b0](https://github.com/vivid-planet/react-admin/commit/5e851b0bbacaf5b59c8f85c1d0eb06ff70390784)

Find ich gut so, dass sollte immer true sein.

Reply:
f9d08a6bba65235b617bd92372232456c5ee054c löst das, bei 'filter-change' wird einfach nie gescrollt.